Xamarin.Essentials/FileSystem/FileSystem.ios.cs

```diff
@@ -176,25 +179,62 @@ internal override Task<Stream> PlatformOpenReadAsync()
         }
     }
 
+    // TODO: Think if we should rename it as UIImage is just an input
     class UIImageFileResult : FileResult
     {
-        readonly UIImage uiImage;
-        NSData data;
-
-        internal UIImageFileResult(UIImage image)
+        internal UIImageFileResult(UIImage image, NSDictionary metadata)
             : base()
         {
-            uiImage = image;
-
-            FullPath = Guid.NewGuid().ToString() + FileSystem.Extensions.Png;
-            FileName = FullPath;
+            FileName = Guid.NewGuid() + FileSystem.Extensions.Jpg;
+            var tempDir = NSFileManager.DefaultManager.GetTemporaryDirectory().Path;
+            FullPath = Path.Combine(tempDir, FileName);
+            SaveImageWithMetadata(image, metadata);
         }
 
         internal override Task<Stream> PlatformOpenReadAsync()
         {
-            data ??= uiImage.AsPNG();
+            Stream fileStream = File.OpenRead(FullPath);
+
+            return Task.FromResult(fileStream);
+        }
+
+        void SaveImageWithMetadata(UIImage image, NSDictionary metadata)
+        {
+            try
+            {
+                var imageData = image.AsJPEG();
```

Comment:
The down side of JPEG is that is compressed and loses transparency. For example, I might pick a png from my downloads folder.

Or maybe this is not the case that it will go down this path?

Reply:
Is it possible to maybe detect the source image and use that? Can a iOS save PNG data?

Reply:
> The down side of JPEG is that is compressed and loses transparency

Yes, that's where the difficulty lies, I myself can not figure out how to do it correctly to keep images and photos original with all the data without compression and loss of transparency.